src/components/PrivateConsumerAccessFields.tsx

```diff
@@ -0,0 +1,508 @@
+  const [courseAutocompleteTypedValue, setCourseAutocompleteTypedValue] = useState('');
+
+  const addInstitutionPrivateConsumerAccess = async () => {
+    await postCurrentUserInstitutionConsumerAccess(values.identifier);
+    if (privateAccessList.findIndex((privateAccess) => privateAccess.subject === user.institution) === -1) {
+      setPrivateAccessList((prevState) => [
+        ...prevState,
+        {
+          subject: user.institution,
+          profiles: [{ name: ResourceReadAccessNames.Institution }],
+        },
+      ]);
+    }
+  };
+
+  useEffect(() => {
+    const getPrivateAccessList = async () => {
+      const resourceReadAccessListResponse = await getResourceReaders(values.identifier);
+      setPrivateAccessList(resourceReadAccessListResponse.data);
+    };
+    getPrivateAccessList();
+  }, [values.identifier]);
+
+  const deleteAccess = async (access: ResourceReadAccess) => {
+    try {
+      let deleteAPISuccessful = false;
+      if (access.profiles[0].name === ResourceReadAccessNames.Institution) {
+        setUpdatingPrivateAccessList(true);
+        await deleteCurrentUserInstitutionConsumerAccess(values.identifier);
+        deleteAPISuccessful = true;
+      } else if (access.profiles[0].name === ResourceReadAccessNames.Person) {
+        setUpdatingPrivateAccessList(true);
+        await deleteAdditionalUserConsumerAccess(values.identifier, access.subject);
+        deleteAPISuccessful = true;
+      } else if (access.profiles[0].name === ResourceReadAccessNames.Course) {
+        setUpdatingPrivateAccessList(true);
+        const tempCourse = access.subject.split(':: ');
+        deleteAPISuccessful = true;
+        //tempCourse[0]: courseCode, tempCourse[1]: institution, tempCourse[2]: year, tempCourse[3]: Season
+        if (tempCourse[0]?.trim().length > 0 && tempCourse[2]?.trim().length > 0 && tempCourse[3]?.trim().length > 0) {
+          await deleteCourseConsumerAccess(
+            values.identifier,
+            tempCourse[0].trim(),
+            tempCourse[2].trim(),
+            tempCourse[3].trim()
+          );
+          deleteAPISuccessful = true;
+        }
+      }
+      if (deleteAPISuccessful) {
+        setPrivateAccessList((prevState) => prevState.filter((accessState) => accessState !== access));
+        setSavePrivateAccessNetworkError(false);
+      }
+    } catch (error) {
+      setSavePrivateAccessNetworkError(true);
+    } finally {
+      setUpdatingPrivateAccessList(false);
+    }
+  };
+
+  const generateChipLabel = (access: ResourceReadAccess): string => {
+    if (access.profiles[0].name === ResourceReadAccessNames.Person) {
+      return access.subject;
+    } else if (access.profiles[0].name === ResourceReadAccessNames.Institution) {
+      return `${t('access.everyone_at')} ${access.subject}`;
+    } else {
+      const courseTemp = access.subject.split(' :: ');
+      if (courseTemp[0]) {
```

Comment:
kanskje bruke et course-objekt her også for å lettere se hva f.ex course[0] står for

Reply:
Fikset